stl/src/stacktrace.cpp

```diff
@@ -0,0 +1,274 @@
+            // Deliberately not calling CoInitialize[Ex]
+            // DbgEng.h API works fine without it. COM initialization may have undesired interference with user's code
+
+            if (SUCCEEDED(DebugCreate(IID_IDebugClient, reinterpret_cast<void**>(&debug_client)))
+                && SUCCEEDED(debug_client->QueryInterface(IID_IDebugSymbols, reinterpret_cast<void**>(&debug_symbols)))
+                && SUCCEEDED(
+                    debug_client->QueryInterface(IID_IDebugControl, reinterpret_cast<void**>(&debug_control)))) {
+                attached = SUCCEEDED(debug_client->AttachProcess(
+                    0, GetCurrentProcessId(), DEBUG_ATTACH_NONINVASIVE | DEBUG_ATTACH_NONINVASIVE_NO_SUSPEND));
+                if (attached) {
+                    debug_control->WaitForEvent(0, INFINITE);
+                }
+            }
+        }
+
+        atexit([] {
+            // "Phoenix singleton" - destroy and set to null, so that can initialize later again
+
+            if (debug_client != nullptr) {
+                if (attached) {
+                    debug_client->DetachProcesses();
+                    attached = false;
+                }
+
+                debug_client->Release();
+                debug_client = nullptr;
+            }
+
+            if (debug_control != nullptr) {
+                debug_control->Release();
+                debug_control = nullptr;
+            }
+
+            if (debug_symbols != nullptr) {
+                debug_symbols->Release();
+                debug_symbols = nullptr;
+            }
+
+            initialize_attempted = false;
+        });
+
+        initialize_attempted = true;
+
+        return debug_symbols != nullptr;
+    }
+
+    size_t get_description(const void* const address, void* str, size_t off, _Stacktrace_string_fill fill) {
+        if (!try_initialize()) {
+            return 0;
+        }
+
+        size_t size          = 20; // a guess, will retry with greater if wrong
+        HRESULT hr           = E_UNEXPECTED;
+        ULONG64 displacement = 0;
+
+        for (;;) {
+            ULONG new_size = 0;
+
+            size_t new_off = string_fill(
+                fill, off + size, str, [address, off, size, &new_size, &hr, &displacement](char* s, size_t) {
+                    hr = debug_symbols->GetNameByOffset(reinterpret_cast<uintptr_t>(address), s + off,
+                        static_cast<ULONG>(size), &new_size, &displacement);
+
+                    return (hr == S_OK) ? off + new_size - 1 : off;
+                });
+
+            if (hr == S_OK) {
+                off = new_off;
+                break;
+            } else if (hr == S_FALSE) {
+                size = new_size; // retry with bigger buffer
+            } else {
+                return off;
+            }
+        }
+
+        if (displacement != 0) {
+            constexpr size_t max_disp_num = std::size("+0x1111222233334444") - 1; // maximum possible line number

```

Comment:
Could this be `"+0x1111222233334444"sv.size()` instead? Ditto below

Reply:
Good point, but I'd want to stand away from `string_view` for this reason:
```C++
// This must be as small as possible, because its contents are
// injected into the msvcprt.lib and msvcprtd.lib import libraries.
// Do not include or define anything else here.
// In particular, basic_string must not be included here.
```

(See #2141 for more precise message)